app/contacts/contact_slack.py

```diff
@@ -0,0 +1,308 @@
+import aiohttp
+import asyncio
+import json
+import os
+import re
+import time
+
+from base64 import b64encode, b64decode
+from collections import defaultdict
+
+from app.utility.base_world import BaseWorld
+
+
+def api_access(func):
+    async def process(*args, **kwargs):
+        async with aiohttp.ClientSession(headers=dict(Authorization='Bearer {}'.format(args[0].key)),
+                                         connector=aiohttp.TCPConnector(verify_ssl=False)) as session:
+            kwargs['session'] = session
+            return await func(*args, **kwargs)
+    return process
+
+
+class Contact(BaseWorld):
+
+    class SlackUpload:
+        def __init__(self, upload_id, filename, num_chunks):
+            self.upload_id = upload_id
+            self.filename = filename
+            self.chunks = [None]*num_chunks
+            self.required_chunks = num_chunks
+            self.completed_chunks = 0
+            self.exported = False
+
+        def add_chunk(self, chunk_index, contents):
+            if self.chunks[chunk_index] is None:
+                self.chunks[chunk_index] = contents
+                self.completed_chunks += 1
+
+        def is_complete(self):
+            return self.completed_chunks == self.required_chunks
+
+        def export_contents(self):
+            self.exported = True
+            return b''.join(self.chunks)
+
+    def __init__(self, services):
+        self.name = 'slack'
+        self.description = 'Use slack for C2'
+        self.file_svc = services.get('file_svc')
+        self.contact_svc = services.get('contact_svc')
+        self.log = self.create_logger('contact_slack')
+        self.key = ''
+        self.channelid = ''
+        self.botid = ''
+
+        # TODO
+        # Stores uploaded file chunks. Maps paw to dict that maps upload ID to SlackUpload object
+        self.pending_uploads = defaultdict(lambda: dict())
+
+    def retrieve_config(self):
+        return self.key
+
+    async def start(self):
+        if await self.valid_config():
+            self.key = self.get_config('app.contact.slack')
+            self.channelid = self.get_config('app.contact.slackchannelid')
+            self.botid = self.get_config('app.contact.slackbotid')
```

Comment:
if you can change this to app.contact.slack.bot_id here and in the default.yml file, that'll keep things more consistent